### ISSUE TITLE
Grammar discrepancy on fn:pin() examples

### ISSUE THREAD

**Issue by johnlumley** [qt4cg/qtspecs#1107]
I'm probably getting the wrong end of the stick, but I can't see how the example for [fn:pin():](https://qt4cg.org/specifications/xpath-functions-40/Overview.html#func-pin)
```
pin(["a","b","c"])?1 => label()?parent => array:foot()
```
meets the current EBNF. (I know bits of this area may be in flux, so this may be just for the record .)
```
pin(["a","b","c"])?1 => label()
```
meets the production for `ArrowExpr`, and the RHS of an `ArrowExpr` is, in this case, an `ArrowStaticFunction` `ArgumentList` pair, which doesn't encompass the subsequent lookup.

But for `LookupExpr` to include the `?parent` requires `PostfixExpr` as its first term, and `PostfixExpr` can only include `ArrowExpr` via `PrimaryExpr/ParenthesizedExpr`, i.e. with brackets.

**Comment by ChristianGruen** [comment]
True. The given query could be rewritten as:

```xquery
pin([ "a", "b", "c"] )?1
! label(.)?parent
! array:foot(.)
```

It’s a common problem with the arrow operator that it can only be used for simple function calls (it’s one reason why I would like to have a value map operator, as motivated in #755).

**Comment by ChristianGruen** [comment]
I’ve added a fix in the PR #1078.
